src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs

```diff
@@ -29,6 +39,36 @@ public sealed class RestartComputerCommand : CommandLineCmdletBase
         /// </summary>
         protected override void BeginProcessing()
         {
+            const string unixRestartCommand = "/sbin/shutdown";
+            const string unixRestartArgs = "-r now";
+            
+            const string macOSRestartCommand = "osascript";
+            const string macOSRestartArgs = @"-e 'tell application ""System Events"" to restart'";
+
+            const string macOSForceRestartCommand = "/sbin/shutdown";
+            const string macOSForceRestartArgs = "-r now";            
+
+            string command;
+            string args;
+
+            if (Platform.IsMacOS)
+            {
+                if (Force.IsPresent)
+                {
+                    command = macOSForceRestartCommand;
+                    args = macOSForceRestartArgs;
+                } 
+                else 
+                {
+                    command = macOSRestartCommand;
+                    args = macOSRestartArgs;  
+                }
+            } 
+            else {
+                command = unixRestartCommand;
+                args = unixRestartArgs;
+            }
+            
```

Comment:
now that you've changed the way shutdown is being called (and adding the more gentle osascript approach, I would be happier if we boosted the tests here, at least to have validation between force and not-force on mac

Reply:
In CI, a test hook will have to be needed to verify intended behavior